### PR TITLE
Reword comments and rename HIR visiting methods.

### PR DESCRIPTION
**Pull request by cjgillot**
Sparked by this discussion in [zulip](https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/Confused.20by.20comment.20on.20.60deep_visit_item_likes_in_module.60)

r? @jyn514 @camsteffen

### CODE REVIEW

File: src/test/ui/dep-graph/dep-graph-struct-signature.stderr [hunk 1]
```diff
@@ -16,12 +16,6 @@ error: no path from `WillChange` to `trait_def`
 LL |     #[rustc_then_this_would_need(trait_def)]
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-error: OK
-  --> $DIR/dep-graph-struct-signature.rs:32:9
-   |
-LL |         #[rustc_then_this_would_need(fn_sig)]
-   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (jyn514) [inline src/test/ui/dep-graph/dep-graph-struct-signature.stderr:23]: this looks like a real regression - I don't see this diagnostic reappearing later in the file.

Comment (cjgillot) [reply src/test/ui/dep-graph/dep-graph-struct-signature.stderr:23]: It's at (new) line 100.

File: compiler/rustc_middle/src/hir/map/mod.rs [hunk 2]
```diff
@@ -588,53 +593,60 @@ impl<'hir> Map<'hir> {
...
         V: Visitor<'hir>,
     {
-        let krate = self.krate();
-        for owner in krate.owners.iter().filter_map(|i| i.as_owner()) {
-            match owner.node() {
-                OwnerNode::Item(item) => visitor.visit_item(item),
-                OwnerNode::ForeignItem(item) => visitor.visit_foreign_item(item),
-                OwnerNode::ImplItem(item) => visitor.visit_impl_item(item),
-                OwnerNode::TraitItem(item) => visitor.visit_trait_item(item),
-                OwnerNode::Crate(_) => {}
```
[the remark this answers is not in the document]
Comment (cjgillot) [reply compiler/rustc_middle/src/hir/map/mod.rs:610]: This only changes the order in which everything is visited, to get closer to `visit_item_likes_in_module`.

File: compiler/rustc_middle/src/hir/map/mod.rs [hunk 3]
```diff
@@ -568,9 +568,14 @@ impl<'hir> Map<'hir> {
...
     pub fn walk_toplevel_module(self, visitor: &mut impl Visitor<'hir>) {
-        let (top_mod, span, hir_id) = self.get_module(CRATE_DEF_ID);
+        self.walk_module(CRATE_DEF_ID, visitor)
+    }
+
+    /// Walks the contents of a module. See also `visit_item_likes_in_module`.
+    pub fn walk_module(self, module: LocalDefId, visitor: &mut impl Visitor<'hir>) {
+        let (top_mod, span, hir_id) = self.get_module(module);
         visitor.visit_mod(top_mod, span, hir_id);
     }
```
[the remark this answers is not in the document]
Comment (cjgillot) [reply compiler/rustc_middle/src/hir/map/mod.rs:580]: Just for consistency.  No problem with removing it though.

[the remark this answers is not in the document]
Comment (jyn514) [reply compiler/rustc_middle/src/hir/map/mod.rs:580]: Yeah, I think I'd prefer to remove this if no one is currently using it. Thanks.

File: compiler/rustc_middle/src/hir/map/mod.rs [hunk 4]
```diff
@@ -588,53 +593,60 @@ impl<'hir> Map<'hir> {
...
 
-    /// Visits all items in the crate in some deterministic (but
-    /// unspecified) order. If you need to process every item,
-    /// and care about nesting -- usually because your algorithm
-    /// follows lexical scoping rules -- then this method is the best choice.
-    /// If you don't care about nesting, you should use the `tcx.hir_crate_items()` query
-    /// or `items()` instead.
+    /// Visits all items in the crate in some deterministic (but unspecified) order. If you need to
+    /// process every item, and don't care about nesting between items then this method is the best
```
Comment (camsteffen) [inline compiler/rustc_middle/src/hir/map/mod.rs:597]: To clarify a bit
```suggestion
    /// process every item, and don't care about visiting nested items in a particular order, then this method is the best
```

File: compiler/rustc_middle/src/hir/map/mod.rs [hunk 5]
```diff
@@ -588,53 +593,60 @@ impl<'hir> Map<'hir> {
...
+        for id in krate.foreign_items() {
+            visitor.visit_foreign_item(self.foreign_item(id));
         }
     }
 
-    /// If you don't care about nesting, you should use the
-    /// `tcx.hir_module_items()` query or `module_items()` instead.
-    /// Please see notes in `deep_visit_all_item_likes`.
-    pub fn deep_visit_item_likes_in_module<V>(self, module: LocalDefId, visitor: &mut V)
+    /// This method is the equivalent of `visit_all_item_likes` but restricted to a single module.
```
Comment (camsteffen) [inline compiler/rustc_middle/src/hir/map/mod.rs:629]: A couple clarifications. It wasn't super obvious to me that nested modules are not included.
```suggestion
    /// This method is the equivalent of `visit_all_item_likes` but restricted to items in a single module.
    /// Nested modules are not visited.
```

File: compiler/rustc_middle/src/hir/map/mod.rs [hunk 6]
```diff
@@ -588,53 +593,60 @@ impl<'hir> Map<'hir> {
...
+    /// choice.  If you do care about this nesting, you should use the
+    /// `tcx.hir().walk_toplevel_module`.
+    ///
+    /// Note that this function will access HIR for all the items in the crate.  If you only need
+    /// to access some of them, it is usually better to manually loop on the iterators provided by
+    /// `tcx.hir_crate_items(())`.
     ///
     /// Please see the notes in `intravisit.rs` for more information.
-    pub fn deep_visit_all_item_likes<V>(self, visitor: &mut V)
+    pub fn visit_all_item_likes<V>(self, visitor: &mut V)
```
Comment (camsteffen) [inline compiler/rustc_middle/src/hir/map/mod.rs:606]: Maybe?
```suggestion
    pub fn visit_all_item_likes_in_crate<V>(self, visitor: &mut V)
```

**Review by camsteffen — COMMENTED** [review]
This is better, thanks! Some minor suggestions.

**Comment by jyn514** [comment]
Sorry for the delay. r=me once you fix the merge conflict.

**Pull request merged**